pkg/cmd/completion.go

```diff
@@ -0,0 +1,36 @@
+package cmd
+
+import (
+	"fmt"
+
+	"github.com/spf13/cobra"
+)
+
+func NewCompletionCmd() *cobra.Command {
+	var cmd = &cobra.Command{
+		Use:                   "completion [bash|zsh|fish|powershell]",
+		Short:                 "Generate completion script",
+		Long:                  "Generate completion script for various shells",
+		DisableFlagsInUseLine: true,
+		ValidArgs:             []string{"bash", "zsh", "fish", "powershell"},
+		Args:                  cobra.ExactValidArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			var err error
+			switch args[0] {
+			case "bash":
+				err = cmd.Root().GenBashCompletion(cmd.OutOrStdout())
+			case "zsh":
+				err = cmd.Root().GenZshCompletion(cmd.OutOrStdout())
+			case "fish":
+				err = cmd.Root().GenFishCompletion(cmd.OutOrStdout(), true)
+			case "powershell":
+				err = cmd.Root().GenPowerShellCompletion(cmd.OutOrStdout())
+			}
+			if err != nil {
+				return fmt.Errorf("error while generating completion script: %s", err.Error())
```

Comment:
Maybe we can have something like this instead ?

```
switch args[0] {
			case "bash":
				return cmd.Root().GenBashCompletion(cmd.OutOrStdout())
			case "zsh":
				return cmd.Root().GenZshCompletion(cmd.OutOrStdout())
			case "fish":
				return cmd.Root().GenFishCompletion(cmd.OutOrStdout(), true)
			case "powershell":
				return cmd.Root().GenPowerShellCompletion(cmd.OutOrStdout())
			}
```

Reply:
I think we could, it was more to log a possible custom error on our side

Reply:
Which kind of error message you can expect there ?

Reply:
Don't really know what could be a possible error here

Reply:
Removed 814eb1a